index/scorch/snapshot_index.go

```diff
@@ -439,21 +439,26 @@ func (i *IndexSnapshot) TermFieldReader(term []byte, field string, includeFreq,
 func (i *IndexSnapshot) allocTermFieldReaderDicts(field string) (
 	tfr *IndexSnapshotTermFieldReader, dicts []segment.TermDictionary) {
 	i.m2.Lock()
+	defer i.m2.Unlock()
```

Comment:
btw, one thing is I have the impression (might be old/outdated) that defer will allocate memory resources underneath the hood for a closure, so if the code's simple enough, or if it's on the "hot million-ops/sec path", i tend to avoid defer and do it the old-school way of explicit unlock's.  Not sure if allocTermFieldReaderDicts() happens enough for this to matter, so just FYI

Reply:
Noted.